### ISSUE TITLE
Unexplained error "implication would create a cycle"

### ISSUE THREAD

**Issue by fturco** [oniony/TMSU#79]
I'm tagging my e-book collection with tmsu and I think I found a bug with the imply command.

Steps to reproduce:
```
$ tmsu init
$ touch x
$ tmsu tag x topic=precalculus topic=mathematics
$ tmsu imply topic=precalculus topic=mathematics
$ touch y
$ tmsu tag y topic=bash topic=programming
$ tmsu imply topic=bash topic=programming
tmsu: cannot add implication of 'topic=bash' to 'topic=programming': implication would create a cycle
```

Expected result: `topic=bash` should imply `topic=programming` as there is no cycle at all

I'm using version 0.6.1 from the AUR.

**Comment by oniony** [comment]
Thanks for reporting this. That does indeed look like a bug. This is
annoying as I thought I had testcases covering this but never mind. I'll
try to get it fixed this weekend.

**Comment by oniony** [comment]
Sorry I've taken so long to look into this issue.

I've just tried to reproduce it on the master branch and can't, suggesting I've already previously fixed this issue. So you could either wait until 0.7 is release (not entirely sure when that will be as I tend to be busy with other stuff over the summer months) or build from source, which is actually not that difficult (instructions in the `COMPILING` file).

**Comment by oniony** [comment]
Further, I can see I fixed this particular issue in 8b45b15.

**Comment by fturco** [comment]
I built version 0.7 as you suggested, and indeed I can't reproduce this bug anymore. I will always try building the latest git version in the future before reporting other bugs. Thank you for fixing the problem!
